### PR TITLE
[feature] pieces - movements

### PR DESCRIPTION
**Pull request by lufercc**
========================
- add utils to get cells for movements [cell movement]
- add utils to get all cells empty for a movement [board movement]
- add method for movement for paw and rook

please take a look and let me your know your comments for change approaches

### CODE REVIEW

**Review by angelmiguelmontalvo — APPROVED** [review]

File: finalProject-2020-q3/code/BoardMovements.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            CellList downRight = GetCellDownRight(piecesOnBoard, position, downLeft);
+            CellList upLeaft = GetCellUpLeaft(piecesOnBoard, position, downRight);
+            CellList response = GetCellUpRight(piecesOnBoard, position, upLeaft);
+            return response;
+        }
+
+        private static CellList GetCellUpLeaft(Cell[,] piecesOnBoard, Cell cell, CellList response)
+        {
+            Cell next = CellMovements.UpLeftDiagonal(piecesOnBoard, cell);
+            if (ValidatCell(next))
```
Comment (jccolina) [inline finalProject-2020-q3/code/BoardMovements.cs]: @lufercc misspelling in ValidatCell

File: finalProject-2020-q3/code/BoardMovements.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            {
+                response.Add(next);
+                return next.IsEmpty() ?
+                    GetCellUpRight(piecesOnBoard, next, response) : response;
+            }
+            return response;
+        }
+
+        private static CellList GetCellDownRight(Cell[,] piecesOnBoard, Cell cell, CellList response)
```
Comment (jccolina) [inline finalProject-2020-q3/code/BoardMovements.cs:51]: @lufercc is it necessary to pass the response as parameter, maybe we could use the 'out' keyword if necessary

File: finalProject-2020-q3/code/CellMovements.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        private static bool ValidatePositionCell (int row, int column)
+        {
+            if (column<8 && column > 0 && row < 8 && row > 0)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private static Cell ReturnCellInPosition(Cell[,] piecesOnBoard, int row, int column)
```
Comment (jccolina) [inline finalProject-2020-q3/code/CellMovements.cs:59]: @lufercc sinc this is returning only piecesOnBoard[row, column]; is it necessary to have a function? the null returned is a value used to validate something?

**Review by jccolina — CHANGES_REQUESTED** [review]

File: finalProject-2020-q3/code/CellMovements.cs (hunk 3 above)
Comment (lufercc) [reply finalProject-2020-q3/code/CellMovements.cs:59]: Yes,  this was created for cases out of boundary, so I'm validating is cell returned is different to null

**Review by jccolina — APPROVED** [review]

**Pull request merged**